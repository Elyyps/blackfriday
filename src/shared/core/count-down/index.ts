export * from "./components/count-down.component";
