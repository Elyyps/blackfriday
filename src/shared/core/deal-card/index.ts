export * from "./deal-card.component";
