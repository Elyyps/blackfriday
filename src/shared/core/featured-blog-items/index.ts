export * from "./featured-blog-items.component";
