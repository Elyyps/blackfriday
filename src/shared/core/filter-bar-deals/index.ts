export * from "./filter-bar-deals.component";
export * from "./container";
