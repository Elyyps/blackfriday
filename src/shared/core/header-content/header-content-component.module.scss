h1 {
  font-weight: bold;
  color: currentColor;
  &.subtitle {
    font-weight: normal;
    margin-top: -16px;
  }
}

.text {
  color: currentColor;
}
