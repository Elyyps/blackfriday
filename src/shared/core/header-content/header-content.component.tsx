import * as React from "react";
import styles from "./header-content-component.module.scss";

export interface IHeaderContentComponentProps {
  backButtonProps?: {};
  breadcrumbItems?: {}[];
  subtitle?: string;
  text?: string;
  title?: string;
}

const HeaderContentComponent = (props: IHeaderContentComponentProps) => {
  const { backButtonProps, breadcrumbItems, title, subtitle, text } = props;

  return (
    <div className={styles["header-content"]}>
      Hier komt de breadcrumb {backButtonProps} {breadcrumbItems}
      {title && <h1>{title}</h1>}
      {subtitle && <h1 className={styles["subtitle"]}>{subtitle}</h1>}
      {text && <span className={styles["text"]}>{text}</span>}
    </div>
  );
};

export { HeaderContentComponent };
