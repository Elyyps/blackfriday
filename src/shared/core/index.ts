export * from "./button";
export * from "./image";
export * from "./usp";
export * from "./icon";
