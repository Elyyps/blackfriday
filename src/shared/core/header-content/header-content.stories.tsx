import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { HeaderContentComponent } from "./header-content.component";
import { headerContent } from "@app/api/core/header-content";

storiesOf("HeaderContent", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => (
    <HeaderContentComponent
      title={headerContent.title}
      subtitle={headerContent.subtitle}
      text={headerContent.text}
      breadcrumbProps={headerContent.breadcrumbProps}
    />
  ));
