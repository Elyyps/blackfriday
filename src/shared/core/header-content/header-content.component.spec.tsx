import React from "react";
import { shallow } from "enzyme";
import { HeaderContentComponent } from "./header-content.component";
import { headerContent } from "@app/api/core/header-content";

describe("[HeaderContent]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(
      <HeaderContentComponent
        title={headerContent.title}
        subtitle={headerContent.subtitle}
        text={headerContent.text}
        breadcrumbProps={headerContent.breadcrumbProps}
      />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
