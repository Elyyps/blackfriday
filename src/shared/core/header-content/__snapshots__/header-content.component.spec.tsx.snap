// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[HeaderContent] should render component without crashing 1`] = `
<div
  className="header-content center"
>
  <BreadcrumbComponent
    backButton={
      Object {
        "title": "Terug",
        "url": "/#",
      }
    }
    links={
      Array [
        Object {
          "title": "Home",
          "url": "/#",
        },
        Object {
          "title": "Blog",
          "url": "/#",
        },
      ]
    }
  />
  <h1>
    Black Friday Winkels
  </h1>
  <h1
    className="subtitle"
  >
    Deelnemende winkels van 2019
  </h1>
  <span
    className="text"
  >
    Bodytext voor extra informatie wanneer nodig
  </span>
</div>
`;
